packages/application-shell/src/hooks/use-applications-menu/use-applications-menu.ts

```diff
@@ -1,60 +1,252 @@
+              return {
+                __typename: 'NavbarMenu',
+                shouldRenderDivider: false,
+                key: menuLink.uriPath,
+                uriPath: menuLink.uriPath,
+                icon: menuLink.icon,
+                labelAllLocales: mapLabelAllLocalesWithDefaults(
+                  menuLink.labelAllLocales,
+                  // There is no `defaultValue`, so we pick the first in the list.
+                  menuLink.labelAllLocales[0].value
+                ),
+                permissions: menuLink.permissions,
+                // @ts-ignore: not defined in schema, as it's only used internally.
+                featureToggle: menuLink.featureToggle ?? null,
+                // @ts-ignore: not defined in schema, as it's only used internally.
+                menuVisibility: menuLink.menuVisibility ?? null,
+                // @ts-ignore: not defined in schema, as it's only used internally.
+                actionRights: menuLink.actionRights ?? null,
+                // @ts-ignore: not defined in schema, as it's only used internally.
+                dataFences: menuLink.dataFences ?? null,
+                submenu: menuLink.submenu.map((submenuLink) => ({
+                  __typename: 'BaseMenu',
+                  key: `${menuLink.uriPath}-${submenuLink.uriPath}`,
+                  uriPath: submenuLink.uriPath,
+                  labelAllLocales: mapLabelAllLocalesWithDefaults(
+                    submenuLink.labelAllLocales
+                  ),
+                  permissions: submenuLink.permissions,
+                  // @ts-ignore: not defined in schema, as it's only used internally.
+                  featureToggle: submenuLink.featureToggle ?? null,
+                  // @ts-ignore: not defined in schema, as it's only used internally.
+                  menuVisibility: submenuLink.menuVisibility ?? null,
+                  // @ts-ignore: not defined in schema, as it's only used internally.
+                  actionRights: submenuLink.actionRights ?? null,
+                  // @ts-ignore: not defined in schema, as it's only used internally.
+                  dataFences: submenuLink.dataFences ?? null,
+                })),
+              };
+            })
+          : [],
+      // @ts-ignore: to suppress `featureToggle` error.
+      appBar:
+        menuKey === 'appBar'
+          ? menuJson.map((menuLink) => ({
+              __typename: 'BaseMenu',
+              key: menuLink.uriPath,
+              uriPath: menuLink.uriPath,
+              labelAllLocales: mapLabelAllLocalesWithDefaults(
+                menuLink.labelAllLocales
+              ),
+              permissions: menuLink.permissions,
+              // @ts-ignore: not defined in schema, as it's only used internally.
+              featureToggle: menuLink.featureToggle ?? null,
+            }))
+          : [],
+    },
+  };
+};
+
 function useApplicationsMenu<Key extends MenuKey>(
```

Comment:
The implementation of the `useApplicationsMenu` changed a bit now:

* it supports loading the menu links from the `window.app.__DEVELOPMENT__.menuLinks`.
* it uses the Apollo cache to "inject" the menu links and let the Apollo query load the data only from the cache (development only).